designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/CustomActionInvokerService.scala

```diff
@@ -63,6 +58,18 @@ class CustomActionInvokerServiceImpl(processRepository: FetchingProcessRepositor
     }
   }
 
-  private def actionError(error: CustomActionError): Future[CustomActionResult] = Future.failed(error)
+  private def actionError(error: Exception with EspError): Future[CustomActionResult] = Future.failed(error)
 
 }
+
+case class CustomActionInvalidStatus(request: CustomActionRequest, stateStatusName: String)
+  extends Exception(s"""Invalid scenario status: $stateStatusName is not allowed for action "${request.name}"""") with BadRequestError
+
+case class CustomActionNonExisting(request: CustomActionRequest)
+  extends Exception(s"""Action "${request.name}" does not exist""") with NotFoundError
+
+case class CustomActionOnFragmentForbidden()
```

Comment:
Use case object instead of parameterless case class

Reply:
Fixed.